src/jobs/distribute_rewards.rs

```diff
@@ -86,7 +113,124 @@ impl DistributeRewardsJob {
                 client.clone(),
             );
 
-            // Preview distribution (no retry for lightweight read operations)
+            // ===== STEP 1: Check snapshot state =====
+            println!("📸 Checking snapshot state...");
+
+            let last_snapshot_timestamp = redistributor_contract.last_snapshot_timestamp().await?;
+            let last_snapshot_block = redistributor_contract.last_snapshot_block_number().await?;
+            let max_age_seconds = redistributor_contract.snapshot_max_age().await?;
+            let current_block = client.get_block_number().await?;
+            let current_timestamp = Self::get_current_timestamp(&client).await?;
```

Comment:
For future reference, we can use multicall https://alloy.rs/guides/multicall/ for all onchain data reads.